metadata-ingestion/src/datahub/ingestion/source/sql/trino.py

```diff
@@ -131,10 +160,42 @@ def _get_columns(self, connection, table_name, schema: str = None, **kw):  # typ
 TrinoDialect._get_columns = _get_columns
 
 
+@functools.lru_cache()
+def get_catalog_connector_name(
+    catalog_name: str, inspector: Inspector
+) -> Optional[str]:
+    if inspector.engine:
+        query = dedent(
+            """
+            SELECT *
+            FROM "system"."metadata"."catalogs"
+        """
+        ).strip()
+        res = inspector.engine.execute(sql.text(query))
+        catalog_connector_dict = {row.catalog_name: row.connector_name for row in res}
+        return catalog_connector_dict.get(catalog_name)
+    return None
+
+
+class ConnectorDetail(PlatformInstanceConfigMixin, EnvConfigMixin):
+    connector_database: Optional[str] = Field(default=None, description="")
+
+
 class TrinoConfig(BasicSQLAlchemyConfig):
     # defaults
     scheme: str = Field(default="trino", description="", hidden_from_docs=True)
 
+    catalog_to_connector_details: Dict[str, ConnectorDetail] = Field(
```

Comment:
let's make sure we're using the terms catalog and connector in a way that's consistent with https://trino.io/docs/current/overview/concepts.html#data-sources

The docs for this are also pretty unclear - terms like "three tier connector" are not commonplace

Reply:
Let me know it its still unclear